django/db/migrations/operations/models.py

```diff
@@ -716,3 +716,79 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
 
     def describe(self):
         return "Change managers on %s" % (self.name, )
+
+
+class IndexOperation(Operation):
+
+    def __init__(self, model_name, index):
+        self.model_name = model_name
+        self.index = index
+
+    def deconstruct(self):
+        kwargs = {
+            'model_name': self.model_name,
+            'index': self.index,
+        }
+        return (
+            self.__class__.__name__,
+            [],
+            kwargs,
+        )
+
+
+class CreateIndex(IndexOperation):
+    """
+    Creates an index for a model.
+    """
+
+    def state_forwards(self, app_label, state):
+        model_state = state.models[app_label, self.model_name.lower()]
+        if not hasattr(self.index, 'model'):
+            self.index.model = state.apps.get_model(app_label, self.model_name)
+        if 'indexes' not in model_state.options:
+            model_state.options['indexes'] = []
+        model_state.options['indexes'].append(self.index)
```

Comment:
Does the ordering of indexes on a database matter? I'm not talking about the order of fields an index applies to, but the indexes itself.